sa/storage-authority.go

```diff
@@ -229,6 +230,73 @@ func (ssa *SQLStorageAuthority) GetLatestValidAuthorization(registrationID int64
 	return ssa.GetAuthorization(auth.ID)
 }
 
+// incrementIP increments the byte within `ip` at `index`, accounting for overflow.
+// Index must be greater than zero and less than len(ip).
+func incrementIP(ip *net.IP, index int) {
+	if (*ip)[index] == 255 {
+		(*ip)[index-1]++
+		(*ip)[index] = 0
+	} else {
+		(*ip)[index]++
+	}
+}
+
+// ipRange returns a range of IP addresses suitable for querying MySQL for the
+// purpose of rate limiting using a range that is inclusive on the lower end and
+// exclusive at the higher end. If ip is an IPv4 address, it returns that address,
+// plus the one immediately higher than it. If ip is an IPv6 address, it applies
+// a /48 mask to it and returns the lowest IP in the resulting network, and the
+// first IP outside of the resulting network.
+func ipRange(ip net.IP) (net.IP, net.IP) {
+	ip = ip.To16()
+	begin := make(net.IP, len(ip))
+	end := make(net.IP, len(ip))
+	// For IPv4 addresses, do a match on exact address, so begin = ip and end =
+	// next higher IP.
+	if ip.To4() != nil {
```

Comment:
Why not just convert every ip to ipv6 and do the work on that? the incrementIP stuff seems very finicky